### ISSUE TITLE
NavigationView menu items list does not have AutomationProperties.Name set

### ISSUE THREAD

**Issue by joshkoon** [microsoft/microsoft-ui-xaml#564]
**Describe the bug**
The NavigationView control does not have AutomationProperties.Name set on the list of menu items, nor can this property be set by consumers of the control

**Steps to reproduce the bug**

1. Add a NavigationView control to an app with a few menu items
2. Activate Narrator and navigate to the list
3. Narrator reads the name of the item, and "1 of X", but there is no context as to what the list is

**Expected behavior**
When using Narrator, users should understand the list is the control's navigation items

**Screenshots**
N/A

**Version Info**

NuGet package version: 
Microsoft.UI.Xaml 2.1.190405004

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (18xxx Latest)                   | Yes   |
| October 2018 Update (17763)   |    |
| April 2018 Update (17134)        |    |
| Fall Creators Update (16299)     |    |
| Creators Update (15063)           |    |
| Anniversary Update (14393)      |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | Yes   |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Additional context**

**Comment by jevansaks** [comment]
Is there a missing TemplateBinding to make AutomationProperties.Name work? @joshkoon would you be willing to contribute a fix? :)

**Comment by jevansaks** [comment]
@joshkoon Sorry for the long delay -- we may have misunderstood this bug when it came in. What is the problem you're trying to solve? NavigationView and the ListView (container) it contains aren't things that narrator stops on so adding names to them doesn't change the read-out for narrator, does it?

@chingucoding you were working on a fix for this -- what changes in the user experience with the change you were making?

**Comment by chingucoding** [comment]
@jevansaks The current experience with NavigationView with narrator is the following: 
User navigates into the NavigationView and the first thing after reading out the Close pane button and the search box, the narrator says : "Item 1 of ..." as it enters the ListView of the NavigationView menu items.

With #1266 , by setting the AutomationProperties.Name property of the NavigationView, the ListView can be announced to the user in a more accessible way. For example, the narrator will then say "Main navigation items Item 1 of ..." when the NavigationView has the accessible name set to "Main navigation items".

**Comment by jevansaks** [comment]
And does it make sense for that name to go onto the ListView? Because there's also a "Window" element for the pane which is nameless right?

**Comment by chingucoding** [comment]
As far as I know (and tested) the "Window" element can not be actually be focuses with narrator. However the list with the menu items can be navigated with the narrator. So I guess it makes sense, but I see that this is debatable.

**Comment by joshkoon** [comment]
The feedback we've received that led to this issue being filed is that the ListView should have an Automation Name to provide the experience @chingucoding describes, where Narrator will read "Main navigation items Item 1 of ...". Without a name, the control does not provide any context to inform the user what the list is.

**Comment by jevansaks** [comment]
Makes sense. We'll go with @chingucoding's fix. Thanks for the discussion!
